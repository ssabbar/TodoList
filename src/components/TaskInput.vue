<template>
  <div >
    <div class="text-center mb-2">
        <label for="new-todo-input">What needs to be done ?</label>
    </div>
    <div class=" mb-3">
      <input class="form-control" id="new-todo-input" v-model="taskName"/>
    </div>
    <div>
      <button class="btn btn-dark form-control" @click="addTask(taskName)" id="addTaskbutton">Add</button>
    </div>

  </div>
</template>

<script>
export default {
  name: 'TaskInput',
  data () {
    return {
      taskName: ''
    }
  },
  methods: {
    addTask (taskName) {
      this.$store.dispatch({
        type: 'addTask',
        taskName
      })
      this.taskName = ''
    }
  }
}

</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
label {
  font-size: 1.5em;
  font-weight: bold;

}
</style>
