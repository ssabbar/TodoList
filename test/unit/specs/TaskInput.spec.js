import Vue from 'vue'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import TaskInput from '@/components/TaskInput'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Task input', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      addTask: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })
  })

  it('call store action "addTask" when click on add button', async () => {
    const wrapper = mount(TaskInput, { store, localVue })
    wrapper.find('input').setValue('Task1')
    await wrapper.find('button').trigger('click')
    expect(actions.addTask).toHaveBeenCalledWith(expect.anything(), { taskName: 'Task1', type: 'addTask' })
  })

  it('renders correctly', () => {
    const wrapper = mount(TaskInput)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
