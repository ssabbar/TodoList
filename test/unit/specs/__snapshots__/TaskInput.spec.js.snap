// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task input renders correctly 1`] = `
<div>
    <div class="text-center mb-2"><label for="new-todo-input">What needs to be done ?</label></div>
    <div class="mb-3"><input id="new-todo-input" class="form-control"></div>
    <div><button id="addTaskbutton" class="btn btn-dark form-control">Add</button></div>
</div>
`;
